services/brig/src/Brig/Team/Util.hs

```diff
@@ -36,3 +36,15 @@ ensurePermissions u t perms = do
     check :: Maybe TeamMember -> Bool
     check (Just m) = and $ hasPermission m <$> perms
     check Nothing  = False
+
+-- | (Some code duplication with 'Galley.API.Teams.ensureNotElevated'.)
+ensurePermissionToAddUser :: UserId -> TeamId -> Permissions -> ExceptT Error AppIO ()
+ensurePermissionToAddUser u t inviteePerms = do
+    m <- lift $ Intra.getTeamMember u t
+    unless (check m) $
+        throwStd insufficientTeamPermissions
+  where
+    check :: Maybe TeamMember -> Bool
+    check (Just m) = hasPermission m AddTeamMember &&
+                     and (hasCopyPermission m <$> (Set.toList $ inviteePerms ^. self))
```

Comment:
Hm.

* Why?

* Is it the same as checking that `self == copy`?

Reply:
does 279146c5ad0fc9550cdae8d9f93f89d2053d89ad answer this?